🌐 Start translations to Tamil, with first translation for `docs/ta/docs/index.md`

arunppsg opened the pull request:
#5434 

In this PR, I initialized tamil translation directory and adding translations for `index.md` file.

arunppsg committed:
index.md translation into tamil

arunppsg commented on docs/az/mkdocs.yml:
```diff
@@ -131,6 +132,8 @@ extra:
   - link: /sv/
     name: sv - svenska
+  - link: /ta/
+    name: ta - Tamil
```
`தமிழ்`  என மாற்றவும்

arunppsg commented on docs/ta/docs/index.md:
```diff
@@ -0,0 +1,463 @@
+
+சுருக்கமாக, அளவுருவியின் வகை, உடல் போன்றவற்றை **ஒரு முறை** செயற்ப்பணி அளவுருவியாக அறிவித்தல்.
+
+அதை செந்தரம் நவீன பைத்தான் வகைகள் மூலமாக செய்ய வேண்டும்.
```
For `standard`, `செந்தரம்` is used but we need a more precise word. Some choices: நிலை, விதிமுறை

FluffyDietEngine replied:
@arunppsg I would suggest not to use prose version of Tamil, as it would need more help for understanding. If you can use the form of Tamil which is in regular use(with simple words), I will be great. I am interested in helping you also.

arunppsg replied:
Updated the usage to simple terms. Hope it is better now!

tiangolo commented:
Awesome, thank you! :rocket: 

And thanks for creating the issue to coordinate translations, I just added it here for Tamil: https://github.com/tiangolo/fastapi/issues/1497

It might be a good idea to make a PR with just the setup, just the files, without the translation, I can approve and merge that right away. And then you can create another PR adding just the translation. That one will take longer because it needs at least 2 people that understand the language to approve it.

arunppsg committed:
patches

arunppsg commented:
> And thanks for creating the issue to coordinate translations, I just added it here for Tamil: #1497

Thank you!

> It might be a good idea to make a PR with just the setup, just the files, without the translation

Added setup in #5564

Elcomerz commented on docs/ta/docs/index.md:
```diff
@@ -0,0 +1,463 @@
+* **குறைந்த பிழை தவறுகள்**: ஏறக்குறைய 40% உருவாக்குனர்களால் வரும் பிழைகளை குறைக்கிறது.
+* **உள்ளுணர்வு**: சிறந்த தொகுப்பி துணை. எங்கும்<abbr title="தானே முடித்தல்"> முடித்தல் </abbr>. தவறு நீக்குதலில் (debugging) குறைந்த நேரம் செலவீட்டல்.
+* **எளியது**: கற்பதற்க்கம் பயன்பாட்டிற்க்கும் எளிமையாக வடிவமைக்கபட்டது.  ஆவணப்படித்தலில் குறைந்த நேரம்.
+* **குட்டி**: கம்மியான குறீயிடு இரட்டிப்பு. ஒவ்வொரு அளவுருவின் (parameter) அறிவிப்பிலும் பண்முக பண்புகூறுகள். குறைந்த பிழை தவறுகள்
```
கம்மியான குறீயிடு இரட்டிப்பு - > குறைவான குறீயிடு(code) நகல்கள்.
குறைந்த பிழை தவறுகள் -> குறைவான பிழைகள்

arunppsg replied:
updated

Elcomerz commented on docs/ta/docs/index.md:
```diff
@@ -0,0 +1,463 @@
+* **உள்ளுணர்வு**: சிறந்த தொகுப்பி துணை. எங்கும்<abbr title="தானே முடித்தல்"> முடித்தல் </abbr>. தவறு நீக்குதலில் (debugging) குறைந்த நேரம் செலவீட்டல்.
+* **எளியது**: கற்பதற்க்கம் பயன்பாட்டிற்க்கும் எளிமையாக வடிவமைக்கபட்டது.  ஆவணப்படித்தலில் குறைந்த நேரம்.
+* **குட்டி**: கம்மியான குறீயிடு இரட்டிப்பு. ஒவ்வொரு அளவுருவின் (parameter) அறிவிப்பிலும் பண்முக பண்புகூறுகள். குறைந்த பிழை தவறுகள்
+* **வீரியம்**: உற்பத்தி ஒட்டதிற்கான குறியீடுகளை பெறுங்கள். கூடுதலாக தானியக்க ஊடாடும் ஆவணம்.
```
production ready code - > உற்பத்திக்கு தயாரான குறியீடு

arunppsg replied:
updated

Elcomerz commented on docs/ta/docs/index.md:
```diff
@@ -0,0 +1,463 @@
+* **எளியது**: கற்பதற்க்கம் பயன்பாட்டிற்க்கும் எளிமையாக வடிவமைக்கபட்டது.  ஆவணப்படித்தலில் குறைந்த நேரம்.
+* **குட்டி**: கம்மியான குறீயிடு இரட்டிப்பு. ஒவ்வொரு அளவுருவின் (parameter) அறிவிப்பிலும் பண்முக பண்புகூறுகள். குறைந்த பிழை தவறுகள்
+* **வீரியம்**: உற்பத்தி ஒட்டதிற்கான குறியீடுகளை பெறுங்கள். கூடுதலாக தானியக்க ஊடாடும் ஆவணம்.
+* **நிலையானது**: திறந்த API அடிப்படையான <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (முன்னர் Swagger என்று அழைக்கப்பட்டது) மற்றும் <a href="https://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>வை சார்ந்த்து (மற்றும் முழுமையாக ஒற்றுமைக்கொண்டது.)
```
This could be Open API instead of திறந்த API.
அடிப்படையான -> அடிப்படையிலான

Open API அடிப்படையிலான <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (முன்னர் Swagger என்று அழைக்கப்பட்டது) மற்றும் <a href="https://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>வை சார்ந்தது (மற்றும் முழுமையாக ஒற்றுமைக்கொண்டது.)

Elcomerz commented on docs/ta/docs/index.md:
```diff
@@ -0,0 +1,463 @@
+
+<a href="https://typer.tiangolo.com" target="_blank"><img src="https://typer.tiangolo.com/img/logo-margin/logo-margin-vector.svg" style="width: 20%;"></a>
+
+நீங்கள் ஒரு வலைய APIக்கு பதிலாக, முணையத்தில் வேலை செய்யும் <abbr title="கட்டளை கொடு இடைமுகப்பு">க.கொ.இ</abbr> (CLI) செயலி செய்கிறிர் என்றால், <a href="https://typer.tiangolo.com/" class="external-link" target="_blank">**Typer**</a>ஐ பார்க்கவும்.
```
முணையத்தில் (terminal)

arunppsg replied:
updated

arunppsg committed:
typos and corrections

FluffyDietEngine commented:
In my opinion, this is a great translation. But there are no words for programming code, computer terminal in tamil, which has been translated with different context here.

For example,
Programming code has been translated as "குறியீடு", which is actually is the word for symbols.

Same with terminal - முனையம், which is a airport terminal.

This translation will create confusion for people who try to read for first time, which contradict the simplicity of original documentation and misleads with different contexts in many places. 

My suggestion would be using English words directly where there is no tamil word available. 

@tiangolo @Elcomerz @gusty1g

tiangolo commented:
Thanks @FluffyDietEngine! Yep, that makes sense to me. Not sure how easy it is to mix English and how strange would it look for people, but I would do that (for example, I would do it in Spanish, use English words for things that don't have an obvious Spanish version).

gusty1g commented:
Sounds reasonable, I will update the PR by using english words for some technical terms.

tiangolo commented:
📝 Docs preview for commit bfc27b94263dfc3dcea79406a9e01cd953fc2f7e at: https://649a23497ba225195de3ab29--fastapi.netlify.app

tiangolo commented:
📝 Docs preview for commit 564d9b48074d1fe02321243e5a6d6d806742d71c at: https://649a2afeec68df1d46e6b311--fastapi.netlify.app

arunppsg committed:
minor fix

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

alejsdev commented:
Thanks for the interest @arunppsg 🥳 

Do you think this one is ready to be merged now with the suggested changes aplied? @Elcomerz @FluffyDietEngine 🤔

FluffyDietEngine commented:
@alejsdev I believe still the [problem](https://github.com/tiangolo/fastapi/pull/5500#issuecomment-1475062330) exists in many places. Would not recommend to merge the PR.

tiangolo commented:
As this PR had requested changes to be applied but has been inactive for a while, it's now going to be closed. But if there's anyone interested, feel free to create a new PR.
